math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -276,163 +1214,179 @@ auto hpt = d.Define("pt", "sqrt(pxs * pxs + pys * pys)[E>200]")
-   using reverse_iterator = typename Impl_t::reverse_iterator;
-   using const_reverse_iterator = typename Impl_t::const_reverse_iterator;
-
-private:
-   Impl_t fData;
+   static constexpr unsigned N = 8;
+   RVec() : Internal::VecOps::SmallVectorImpl<T>(N) {}
 
-public:
-   // constructors
-   RVec() {}
+   ~RVec()
+   {
+      if (this->Owns) {
+         // Destroy the constructed elements in the vector.
+         this->destroy_range(this->begin(), this->end());
+      }
+   }
 
-   explicit RVec(size_type count) : fData(count) {}
+   explicit RVec(size_t Size, const T &Value = T()) : Internal::VecOps::SmallVectorImpl<T>(N)
+   {
+      this->assign(Size, Value);
+   }
 
-   RVec(size_type count, const T &value) : fData(count, value) {}
+   template <typename ItTy,
+             typename = typename std::enable_if<std::is_convertible<
+                typename std::iterator_traits<ItTy>::iterator_category, std::input_iterator_tag>::value>::type>
+   RVec(ItTy S, ItTy E) : Internal::VecOps::SmallVectorImpl<T>(N)
+   {
+      this->append(S, E);
+   }
 
-   RVec(const RVec<T> &v) : fData(v.fData) {}
+   RVec(std::initializer_list<T> IL) : Internal::VecOps::SmallVectorImpl<T>(N) { this->assign(IL); }
 
-   RVec(RVec<T> &&v) : fData(std::move(v.fData)) {}
+   RVec(const RVec &RHS) : Internal::VecOps::SmallVectorImpl<T>(N)
+   {
+      if (!RHS.empty())
+         Internal::VecOps::SmallVectorImpl<T>::operator=(RHS);
+   }
 
-   RVec(const std::vector<T> &v) : fData(v.cbegin(), v.cend()) {}
+   RVec &operator=(const RVec &RHS)
+   {
+      Internal::VecOps::SmallVectorImpl<T>::operator=(RHS);
+      return *this;
+   }
 
-   RVec(pointer p, size_type n) : fData(n, T(), ROOT::Detail::VecOps::RAdoptAllocator<T>(p)) {}
+   RVec(RVec &&RHS) : Internal::VecOps::SmallVectorImpl<T>(N)
+   {
+      if (!RHS.empty())
+         Internal::VecOps::SmallVectorImpl<T>::operator=(::std::move(RHS));
+   }
 
-   template <class InputIt>
-   RVec(InputIt first, InputIt last) : fData(first, last) {}
+   RVec(Internal::VecOps::SmallVectorImpl<T> &&RHS) : Internal::VecOps::SmallVectorImpl<T>(N)
+   {
+      if (!RHS.empty())
+         Internal::VecOps::SmallVectorImpl<T>::operator=(::std::move(RHS));
+   }
 
-   RVec(std::initializer_list<T> init) : fData(init) {}
+   RVec(const std::vector<T> &RHS) : RVec(RHS.begin(), RHS.end()) {}
 
-   // assignment
-   RVec<T> &operator=(const RVec<T> &v)
+   RVec &operator=(RVec &&RHS)
    {
-      fData = v.fData;
+      Internal::VecOps::SmallVectorImpl<T>::operator=(::std::move(RHS));
       return *this;
    }
 
-   RVec<T> &operator=(RVec<T> &&v)
+   RVec(T* p, size_t n) : Internal::VecOps::SmallVectorImpl<T>(N)
    {
-      std::swap(fData, v.fData);
-      return *this;
+      this->BeginX = p;
+      this->fSize = n;
+      this->Capacity = n;
+      this->Owns = false;
    }
 
-   RVec<T> &operator=(std::initializer_list<T> ilist)
+   RVec &operator=(Internal::VecOps::SmallVectorImpl<T> &&RHS)
```

Comment:
Here, too, I'd use `RVec` as argument type (unless I'm missing something)

Reply:
Solved by merging `RVec` and `SmallVectorImpl` (which has other advantages too)